src/Disks/IDisk.h

```diff
@@ -243,6 +243,11 @@ class IDisk : public Space
     /// Remove file or directory with all children. Use with extra caution. Throws exception if file doesn't exists.
     virtual void removeRecursive(const String & path) = 0;
 
+    /// Remove file or directory with all children. Use with extra caution. Throws exception if file doesn't exists.
+    /// Differs from removeRecursive for S3/HDFS disks
+    /// Removes files every time limit is reached and reduces memory consumption.
```

Comment:
"every time limit is reached" is a little bit confusing. Maybe "Limit the number of removing files in batches to prevent high memory consumption"